### ISSUE TITLE
GiftedChat Clears Input Text on Toggle Action Animation

### ISSUE THREAD

**Issue by Falicitas** [FaridSafi/react-native-gifted-chat#2543]
#### Issue Description

When I toggle the action bar (which triggers an animation) while using react-native-gifted-chat, the input text in the TextInput is cleared. Upon logging, I observed that toggling the action bar and refocusing the TextInput triggers a large number of onInputTextChanged calls, but I am unsure of the cause.

#### Steps to Reproduce / Code Snippets

Code Snippets:

```javascript
import React, {useRef, useState} from 'react';
import {Animated, StyleSheet, TextInput, View} from 'react-native';
import {Text} from 'react-native-paper';
import {GiftedChat} from 'react-native-gifted-chat';
import AccessoryBar from '@/features/chat-list/AccessoryBar.tsx';

const Chat = () => {
  const [chatText, setText] = useState(''); // Manage text state
  const textInputRef = useRef<TextInput>(null);
  console.log(chatText);
  // ...
  const [actionsVisible, setActionsVisible] = useState(false);
  const animatedHeight = useRef(new Animated.Value(0)).current;
  const toggleActionsBar = () => {
    setActionsVisible(!actionsVisible);
    Animated.timing(animatedHeight, {
      toValue: actionsVisible ? 0 : 150,
      duration: 300,
      useNativeDriver: false,
    }).start();
  };
  const renderAccessory = props => {
    return (
      <AccessoryBar
        {...props}
        setText={setText}
        toggleActionsBar={toggleActionsBar}
      />
    );
  };

  return (
    <View style={[styles.fill, styles.content]}>
      <Animated.View style={{flex: 1, marginBottom: animatedHeight}}>
        <GiftedChat
          text={chatText} // Bind input text
          onInputTextChanged={newText => {
            console.log('reach here X');
            setText(newText);
          }}
          textInputRef={textInputRef}
          renderAccessory={renderAccessory} // (Function) - Custom second line of actions below the message composer
        />
      </Animated.View>
      <Animated.View style={[styles.actionsBar, {height: animatedHeight}]}>
        <Text style={styles.actionsBarText}>This is the Actions Bar</Text>
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  fill: {
    flex: 1,
  },
  content: {
    backgroundColor: '#F8FAFA',
  },
  actionsBar: {
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: '#ddd',
    justifyContent: 'center',
    alignItems: 'center',
  },
  actionsBarText: {
    fontSize: 16,
    color: '#333',
  },
});

// AccessoryBar
// ...
<Button
  onPress={() => {
    handleBlur();
    toggleActionsBar();
  }}
  name="e147"
  color={theme.colors.onTertiaryContainer}
/>
```

video: 

https://github.com/user-attachments/assets/79e7c695-2ad6-4017-b84c-13e5626dfc8f

#### Expected Results

1. When the ActionBar is expanded, the text in the TextInput should be retained and not cleared.

2. The container height should not be reset. See next comment for more details.

#### Additional Information

* Nodejs version: v20.17.0
* React version: 18.2.0
* React Native version: 0.74.3
* react-native-gifted-chat version: ^2.6.2
* Platform(s) (iOS, Android, or both?): both
* TypeScript version: 5.0.4

**Comment by Falicitas** [comment]
I attempted to solve this issue by passing the text value and its update method through textInputProps. This approach successfully prevents the text from being cleared; however, it introduced another issue: the input container height collapses.

Here is the modified code:

```javascript
<GiftedChat
  // text={chatText} // Bind input text
  // onInputTextChanged={newText => {
  //   console.log('reach here X');
  //   setText(newText);
  // }}
  textInputRef={textInputRef}
  textInputProps={{
    onFocus: handleFocus,
    value: chatText,
    onChangeText: setText,
  }}
/>
```

video:


https://github.com/user-attachments/assets/5ed436ee-4d24-4610-aa0e-f1795c967a08

**Comment by Falicitas** [comment]
Hi Fukemy, thank you for following up on this issue.

I have roughly reproduced the scenario and removed the text state, but the problem still persists. Any ideas on what I should test next?

<img width="845" alt="image" src="https://github.com/user-attachments/assets/9c935dff-4d31-408d-a731-29635de583ce">

https://github.com/user-attachments/assets/248f6cf6-51d7-4bce-a974-ed5340b1acb4

(Note: The fact that sending text doesn't clear the composer is not an issue—it's just for the purpose of reproducing the problem, so I didn't bind it to the text state.)

**Comment by Falicitas** [comment]
Hi @fukemy, sorry for late reply too. I guess I should set up email notifications for GitHub so I don't miss anything in the future.

I’ve picked up this task again after you mentioned using `textInputRef.current?.clear()`. I’ve been thinking that it might be improper to use `Animated.View` the way I am, as it may causes the child component to rerender. To avoid clearing the text, I commented out all the `textInputRef.clear()` code, but that still caused the height of the composer (InputToolBar) to reset to a single line.

After that, I tried switching from `Animated` to `Reanimated` and used similar techniques like `withTiming`, but things actually got worse—the `ActionBar` was in the wrong initial position. I noticed my code was using `flex=1` to stretch the view to full size and then expanding `marginBottom` with `AnimatedHeight` when toggling the `ToggleActionButton`. I ended up ditching the `marginBottom` approach and tried a few hacky solutions, eventually using `transform` to roughly achieve the effect I wanted.

I’m not sure if I fully understand the fundamental principles behind `Animated` (maybe related to different between JS thread and UI main thread?), but this might be enough to close the issue for now.


https://github.com/user-attachments/assets/cd2b5cc7-54e5-48f9-8850-7770ddfa5b06
